src/components/shared/thread/ActivityGraphCanvas.js

```diff
@@ -0,0 +1,224 @@
+// @flow
+
+import * as React from 'react';
+import { computeActivityGraphFills } from './ActivityGraphFills';
+import { timeCode } from '../../../utils/time-code';
+import { mapCategoryColorNameToStyles } from '../../../utils/colors';
+
+import type {
+  Thread,
+  Milliseconds,
+  SelectedState,
+  IndexIntoSamplesTable,
+  CategoryList,
+} from 'firefox-profiler/types';
+
+import type { CategoryDrawStyles } from './ActivityGraphFills';
+
+type CanvasProps = {|
+  +className: string,
+  +trackName: string,
+  +fullThread: Thread,
+  +interval: Milliseconds,
+  +rangeStart: Milliseconds,
+  +rangeEnd: Milliseconds,
+  +samplesSelectedStates: null | SelectedState[],
+  +treeOrderSampleComparator: (
+    IndexIntoSamplesTable,
+    IndexIntoSamplesTable
+  ) => number,
+  +categories: CategoryList,
+  +passFillsQuerier: any,
```

Comment:
These will need to be typed correctly, rather than using an `any` type.

https://flow.org/en/docs/types/functions/#toc-function-types

Reply:
Thanks for the link. Really helped in typing them correctly